libkbfs/data_types.go

```diff
@@ -941,3 +941,8 @@ func SessionInfoFromProtocol(session keybase1.Session) (SessionInfo, error) {
 		VerifyingKey:   verifyingKey,
 	}, nil
 }
+
+// NodeMetadata has metadata about a node needed for higher level operations.
+type NodeMetadata struct {
```

Comment:
Maybe we can just put the entire `DirEntry` in here?  I can imagine it might be useful for debugging to get the block ID/refnonce, encrypted size, key gen, data version, etc.  What do you think?